### PR TITLE
Multiple runner support

### PR DESCRIPTION
**Pull request by ggggg**
Codetierlist will now have support for multiple runners

### CODE REVIEW

File: backend/src/common/runner/index.ts [hunk 1]
```diff
@@ -144,10 +150,13 @@ export const queueJob = (job: Job) : Promise<JobResult> => {
...
 
-const create_images = () => {
-    console.log("creating images");
-    const img = 'python';
-    const img_ver = '3.10.11';
+// Add new images here
+export const images : RunnerImage[] = [
+    {image: 'python', image_version: '3.10.11'},
+    {image: 'python', image_version: '3.12.1'}
+];
```
Comment (ggggg) [inline backend/src/common/runner/index.ts:157]: @DakshChan this is how we will add new images

File: frontend/pages/courses/[courseID]/admin/create_assignment/index.tsx [hunk 2]
```diff
@@ -52,26 +71,43 @@ export default function Page(): JSX.Element {
...
+                                <Option key={`${image}/${version}`}
+                                    text={`${image}/${version}`}
+                                    value={JSON.stringify({image, image_version: version})}>
+                                    {image}/{version}
+                                </Option>
+                            )}
+                        </OptionGroup>)}
+                    </Dropdown>
 
-                    <br /><br />
```
Comment (logonoff) [inline frontend/pages/courses/[courseID]/admin/create_assignment/index.tsx:74]: use CSS

File: frontend/pages/courses/[courseID]/admin/create_assignment/index.tsx [hunk 3]
```diff
@@ -52,26 +71,43 @@ export default function Page(): JSX.Element {
...
-                    <Label htmlFor="dueDate">Due Date:</Label><br />
-                    <Input type="datetime-local" id="dueDate" name="dueDate"
+                    <Label htmlFor="dueDate">Due Date:</Label><br/>
+                    <Input required type="datetime-local" id="dueDate"
+                        name="dueDate"
                         value={dueDate.toISOString().slice(0, -8)}
-                        onChange={e => setDueDate(new Date(e.target.value))} />
+                        onChange={e => setDueDate(new Date(e.target.value))}/><br/>
+
+                    <Label htmlFor="runner">Runner image:</Label><br/>
```
Comment (logonoff) [inline frontend/pages/courses/[courseID]/admin/create_assignment/index.tsx:94]: maybe add an explainer

**Review by logonoff — CHANGES_REQUESTED** [review]

File: frontend/pages/courses/[courseID]/admin/create_assignment/index.tsx (hunk 2 above)
Comment (ggggg) [reply frontend/pages/courses/[courseID]/admin/create_assignment/index.tsx:74]: This was there before

File: frontend/pages/courses/[courseID]/admin/create_assignment/index.tsx (hunk 3 above)
Comment (ggggg) [reply frontend/pages/courses/[courseID]/admin/create_assignment/index.tsx:94]: like a tooltip?

Comment (logonoff) [reply frontend/pages/courses/[courseID]/admin/create_assignment/index.tsx:94]: that works

File: frontend/pages/courses/[courseID]/admin/create_assignment/index.tsx (hunk 2 above)
Comment (logonoff) [reply frontend/pages/courses/[courseID]/admin/create_assignment/index.tsx:74]: worry werid

**Review by logonoff — APPROVED** [review]

**Pull request merged**